examples/6field-simple/elm_6f.cxx

```diff
@@ -12,1677 +12,1682 @@
+    output.write("Jysep1_1 = {:d}   Grid number = {:e}\n", int(Jysep), Grid_NX);
+
+    if (Jysep > 0.) { // for single null geometry
+
+      BoutReal Jxsep, Jysep2;
+      mesh->get(Jxsep, "ixseps1");
+      mesh->get(Jysep2, "jyseps2_2");
+
+      for (int jx = 0; jx < mesh->LocalNx; jx++) {
+        BoutReal mgx = mesh->GlobalX(jx);
+        BoutReal xgrid_num = (Jxsep + 1.) / Grid_NX;
+        // output.write("mgx = {:e} xgrid_num = {:e}\n", mgx);
+        for (int jy = 0; jy < mesh->LocalNy; jy++) {
+          int globaly = mesh->getGlobalYIndex(jy);
+          // output.write("local y = {:d};   global y: {:d}\n", jy, globaly);
+          if (mgx > xgrid_num || (globaly <= int(Jysep) - 2)
+              || (globaly > int(Jysep2) + 2))
+            mgx = xgrid_num;
+          BoutReal rlx = mgx - n0_center;
+          BoutReal temp = exp(rlx / n0_width);
+          BoutReal dampr = ((temp - 1.0 / temp) / (temp + 1.0 / temp));
+          result(jx, jy) = 0.5 * (1.0 - dampr) * n0_height + n0_ave;
+        }
+      }
+    } else { // circular geometry
+      for (int jx = 0; jx < mesh->LocalNx; jx++) {
+        BoutReal mgx = mesh->GlobalX(jx);
+        BoutReal xgrid_num = Grid_NXlimit / Grid_NX;
+        if (mgx > xgrid_num)
           mgx = xgrid_num;
         BoutReal rlx = mgx - n0_center;
         BoutReal temp = exp(rlx / n0_width);
         BoutReal dampr = ((temp - 1.0 / temp) / (temp + 1.0 / temp));
-        result(jx, jy) = 0.5 * (1.0 - dampr) * n0_height + n0_ave;
+        for (int jy = 0; jy < mesh->LocalNy; jy++)
+          result(jx, jy) = 0.5 * (1.0 - dampr) * n0_height + n0_ave;
       }
     }
-  } else { // circular geometry
-    for (int jx = 0; jx < mesh->LocalNx; jx++) {
-      BoutReal mgx = mesh->GlobalX(jx);
-      BoutReal xgrid_num = Grid_NXlimit / Grid_NX;
-      if (mgx > xgrid_num)
-        mgx = xgrid_num;
-      BoutReal rlx = mgx - n0_center;
-      BoutReal temp = exp(rlx / n0_width);
-      BoutReal dampr = ((temp - 1.0 / temp) / (temp + 1.0 / temp));
-      for (int jy = 0; jy < mesh->LocalNy; jy++)
-        result(jx, jy) = 0.5 * (1.0 - dampr) * n0_height + n0_ave;
-    }
+
+    mesh->communicate(result);
+
+    return result;
   }
 
-  mesh->communicate(result);
+  // Parallel gradient along perturbed field-line
+  Field3D Grad_parP(const Field3D& f, CELL_LOC loc = CELL_DEFAULT) {
+    TRACE("Grad_parP");
 
-  return result;
-}
+    Field3D result;
 
-int physics_init(bool restarting) {
-  bool noshear;
+    if (parallel_lagrange || parallel_project) {
+      // Moving stencil locations
 
-  // Get the metric tensor
-  Coordinates* coord = mesh->getCoordinates();
+      ASSERT2((not mesh->StaggerGrids) or loc == CELL_DEFAULT or loc == f.getLocation());
 
-  output.write("Solving high-beta flute reduced equations\n");
-  output.write("\tFile    : {:s}\n", __FILE__);
-  output.write("\tCompiled: {:s} at {:s}\n", __DATE__, __TIME__);
+      Field3D fp, fm; // Interpolated on + and - y locations
```

Comment:
warning: multiple declarations in a single statement reduces readability [readability-isolate-declaration]

```cpp
Field3D fp, fm; // Interpolated on + and - y locations
^~~~~~~~~~~~~~~
```